### PR TITLE
Add http proxy

### PR DESCRIPTION
**Pull request by AmberCharitos**
For deployment in staging or production environment it needs to be possible to configure the `HTTP_PROXY` and `HTTPS_PROXY` values.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -214,6 +216,9 @@ def _validate_config_params(self):
             raise ValueError(
                 f"config: invalid self-registration role {default_role!r}"
             )
+        for key, value in self.config.items():
+            if value == "":
```
Comment (kelkawi-a) [inline src/charm.py:220]: Won't this force users to include an `http_proxy` config, even if they don't need one?

File: config.yaml [hunk 2]
```diff
@@ -125,3 +125,11 @@ options:
...
     type: string
     default: Public
+  http-proxy:
+    description: |
+      The HTTP_PROXY value.
+    type: string
+  https-proxy:
+    description: |
+      The HTTPS_PROXY value.
+    type: string
```
Comment (kelkawi-a) [inline config.yaml:135]: Do we also need a `no-proxy` config?

File: src/charm.py (hunk 1 above)
Comment (AmberCharitos) [reply src/charm.py:220]: Nope, it checks for empty strings. The default value is and can remain `None`.

Comment (kelkawi-a) [reply src/charm.py:220]: Are we sure the default for config strings is `None` and not `""`?

Comment (AmberCharitos) [reply src/charm.py:220]: I'm fairly certain, have you seen a scenario when this is otherwise?

Comment (AmberCharitos) [reply src/charm.py:220]: @kelkawi-a I believe you are setting the default to `""` in your charms with this `default: ""` but if you do not set anything then I believe the value is `None`.

Comment (kelkawi-a) [reply src/charm.py:220]: I just did a quick test on a charm I have running by doing `juju config <charm_name> --reset <config_key>`, which reset the value back to `""`.

Comment (AmberCharitos) [reply src/charm.py:220]: Is the default value for that config paramter set to be `""` as I described above?

File: config.yaml (hunk 2 above)
Comment (AmberCharitos) [reply config.yaml:135]: We don't need it for our current use cases and I'm not entirely sure if this is supported by Superset - but let me check.

File: src/charm.py (hunk 1 above)
Comment (kelkawi-a) [reply src/charm.py:220]: Ah you're right. Nevermind me then.

File: config.yaml (hunk 2 above)
Comment (AmberCharitos) [reply config.yaml:135]: I have added a `no-proxy` value.

**Review by kelkawi-a — APPROVED** [review]

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -109,6 +109,9 @@ def test_ready(self):
                         "OAUTH_DOMAIN": None,
                         "OAUTH_ADMIN_EMAIL": "admin@superset.com",
                         "SELF_REGISTRATION_ROLE": "Public",
+                        "HTTP_PROXY": None,
```
Comment (kelkawi-a) [inline tests/unit/test_charm.py:112]: Did you want to add any values here to make sure they come through as part of the test?

**Review by mertalpt — APPROVED** [review]
Except for Ali's comments, LGTM.

**Pull request merged**